app/helpers/panel_helper.rb

```diff
@@ -1,68 +1,60 @@
 # frozen_string_literal: true
 
 module PanelHelper
-  def panel_list
-    [
-      {
-        id: :admin,
+  def panel_list(current_user = nil)
+    panel_pages = PanelController.panel_pages
+    {
+      admin: {
         name: 'New Admin panel',
-        url: Rails.application.routes.url_helpers.panel_admin_path,
+        pages: panel_pages.values,
       },
-      {
-        id: :staff,
+      staff: {
         name: 'Staff panel',
-        url: Rails.application.routes.url_helpers.panel_staff_path,
+        pages: [],
```

Comment:
The pages are empty just because they are not moved to this format completely. This will work even with empty pages list because these will rely on panel#staff, and not panel#index.